sdk/client-core/core/src/main/java/io/clientcore/core/http/client/DefaultHttpClient.java

```diff
@@ -194,57 +199,72 @@ private Response<?> receiveResponse(HttpRequest httpRequest, HttpURLConnection c
         }
 
         if (isTextEventStream(responseHeaders)) {
-            ServerSentEventListener listener = httpRequest.getServerSentEventListener();
+            ServerSentEventListener listener = options.getServerSentEventListener();
+            // REMOVE, once confirmed
+            if (listener == null) {
+                listener = httpRequest.getServerSentEventListener();
+            }
```

Comment:
What is this comment suggesting?

Reply:
Just what we talked in the meeting, wanted to confirm if we move setting the `ServerSentEventListener` to `RequestOptions` rather than `HttpRequest`

Reply:
oh, got it. Yep, it makes sense to elevate it to RequestOptions in some form.